### PR TITLE
Add Entry Statistics on top of QC View

### LINKED ISSUE #1183: Entries QA: Add pane on top with stats
Add:
1. Total number of leads
2. Total number of entries
    2.1 Verified
    2.2 Unverified
3. Number of unique authoring organizations by organization type
4. Number of publishers

### PR DESCRIPTION
**Pull request by samshara**
depends on https://github.com/the-deep/server/pull/566
fixes https://github.com/the-deep/client/issues/1183

### Note: the entry stats may change due to #1177 which is implemented by #1194. The entry stats thus changed won't be shown by the solution implemented.

### CODE REVIEW

**Comment by AdityaKhatri** [comment]
@samshara , I think we should pull the stats again after the lead is edited. I just merged the PR you mentioned.

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    className={styles.stats}
+                    renderer={EntryStat}
+                    data={statsList}
+                    keySelector={statsKeySelector}
+                    rendererParams={statsRendererParams}
+                />
+            </div>
+        );
+    }
+    return (null);
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx]: Better to use `if (!stats) { return null; }` to short-circuit

File: src/views/Entries/QualityControl/index.tsx [hunk 2]
```diff
@@ -116,56 +121,86 @@ function QualityControl(props: Props) {
...
             processedFilters,
             activePage,
             selected,
         ],
     );
 
-    const handleEntryDelete = React.useCallback((entryId) => {
+    const handleEntryEdit = useCallback(() => {
+        getEntries();
+    }, [getEntries]);
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/index.tsx]:
```suggestion
const handleEntryEdit = getEntires;
```

File: src/views/Entries/QualityControl/index.tsx [hunk 3]
```diff
@@ -176,15 +211,13 @@ function QualityControl(props: Props) {
...
-        });
-        setEntries(patchedEntries);
-    }, [entries]);
+    const handleLeadEdit = useCallback(() => {
+        getEntries();
+    }, [getEntries]);
+
+    const handleVerificationChange = useCallback(() => {
+        getEntries();
+    }, [getEntries]);
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/index.tsx]:
```suggestion
   const handleLeadEdit = getEntries;
   const handleVerificationChange = getEntries;
```

**Review by tnagorra — APPROVED** [review]

File: src/views/Entries/QualityControl/index.tsx [hunk 4]
```diff
@@ -160,6 +160,7 @@ function QualityControl(props: Props) {
         onSuccess: (response) => {
             setEntries(response.results);
             setStats(response.summary);
+            console.info(response.summary);
```
Comment (AdityaKhatri) [inline src/views/Entries/QualityControl/index.tsx]: Remove this

Comment (frozenhelium) [reply src/views/Entries/QualityControl/index.tsx]: We do not comment on outdated changes

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    orgTypeCount.forEach((orgType) => {
+        statsList.push({
+            id: String(orgType.org.id),
+            title: orgType.org.shortName ?? orgType.org.title,
+            value: orgType.count,
+        });
+    });
+
+    const max = Math.max(...statsList.map(d => d.value || 0));
```
Comment (AdityaKhatri) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx]: @frozenhelium , please memoize these calculations

File: src/views/Entries/QualityControl/index.tsx [hunk 6]
```diff
@@ -116,56 +121,112 @@ function QualityControl(props: Props) {
...
+    const handleEntryEdit = useCallback((updatedEntry) => {
+        setEntries(oldEntries => (
+            oldEntries.map((entry) => {
+                if (entry.id === updatedEntry.id) {
+                    return ({
+                        ...updatedEntry,
+                        lead: entry.lead,
+                    });
+                }
+                return entry;
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/index.tsx]: Even though this works, we should use immer or splice

File: src/views/Entries/QualityControl/index.tsx [hunk 7]
```diff
@@ -116,56 +121,112 @@ function QualityControl(props: Props) {
...
+    const handleLeadEdit = useCallback((updatedLead) => {
+        getEntriesWithStats();
+        setEntries(oldEntries => (
+            oldEntries.map((entry) => {
+                if (entry.lead?.id === updatedLead.id) {
+                    return ({
+                        ...entry,
+                        lead: updatedLead,
+                    });
+                }
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/index.tsx]: Even though this works, we should either user immer or splice

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    title: string;
+    value: number;
+    max: number;
+    className?: string;
+}
+
+function EntryStat({
+    title,
+    value = 0,
+    max,
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx:36]: Should we check if max is zero? before division?

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+    return (
+        <div
+            className={styles.stat}
+            style={{
+                filter: `grayscale(${100 - saturation}%)`,
+            }}
+        >
+            <div className={styles.value}>
+                {value}
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx]: Should we not use Numeral?

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+            value: v,
+        }));
+
+        orgTypeCount.forEach((orgType) => {
+            statsList.push({
+                id: String(orgType.org.id),
+                title: orgType.org.shortName ?? orgType.org.title,
+                value: orgType.count,
+            });
+        });
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx]: We should use `map` instead.

File: src/views/Entries/QualityControl/EntriesStats/index.tsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        return statsList;
+    }, [staticStats, orgTypeCount]);
+
+    const max = useMemo(() => (
+        Math.max(...statsList.map(d => d.value || 0))
```
Comment (tnagorra) [inline src/views/Entries/QualityControl/EntriesStats/index.tsx]: We should include a default max value.
Also, we should prefer `??` if it is available

File: src/views/Entries/QualityControl/EntriesStats/index.tsx (hunk 8 above)
Comment (AdityaKhatri) [reply src/views/Entries/QualityControl/EntriesStats/index.tsx:36]: No its fine

**Pull request merged**